Test/CaptureTest.php

```diff
@@ -0,0 +1,214 @@
+
+        $responses = [
+          $response,
+        ];
+        $client = $this->getClient($responses);
+
+        // Get Rule Manager
+        $manager = $client->getCaptureManager();
+        $response = $manager->add($this->captures);
+
+        // Check for basic fields
+        $this->assertEquals($response->getErrors(), null);
+        $this->assertEquals($response->getTouchIdentifier(), 'my-custom-touch-identifier');
+        $this->assertEquals($response->getIdentity(), 'my-custom-identity-string');
+        $this->assertEquals($response->getIdentitySource(), 'php-unit-test');
+
+        $this->assertEquals($response->getMatchedSegments()[0]->getId(), 'segment-1');
+        $this->assertEquals($response->getMatchedSegments()[0]->getName(), 'Segment 1');
+        $this->assertEquals($response->getMatchedSegments()[0]->getDescription(), 'First Segment for the unit test');
+
+        $this->assertEquals($response->getMatchedSegments()[1]->getId(), 'segment-2');
+        $this->assertEquals($response->getMatchedSegments()[1]->getName(), 'Segment 2');
+        $this->assertEquals($response->getMatchedSegments()[1]->getDescription(), 'Second Segment for the unit test');
+    }
+
+    public function testCaptureAddLiftWebError()
+    {
+
+        // Setup
+        $data = [
+            'errors' => [
+                [
+                    'code' => '400',
+                    'message' => 'Resource had an internal error.',
+                ],
+            ],
+            'touch_identifier' => 'my-custom-touch-identifier',
+            'identity' => 'my-custom-identity-string',
+            'identity_source' => 'php-unit-test',
+        ];
+
+        $response = new Response(200, [], json_encode($data));
+        $responses = [
+          $response,
+        ];
+
+        $client = $this->getClient($responses);
+
+        // Get Rule Manager
+        $manager = $client->getCaptureManager();
+        $response = $manager->add($this->captures);
+
+        $this->assertEquals($response->getTouchIdentifier(), 'my-custom-touch-identifier');
+        $this->assertEquals($response->getIdentity(), 'my-custom-identity-string');
+        $this->assertEquals($response->getIdentitySource(), 'php-unit-test');
+        $this->assertEquals($response->getErrors()[0]->getCode(), '400');
+        $this->assertEquals($response->getErrors()[0]->getMessage(), 'Resource had an internal error.');
+    }
+
+    public function testCaptureAddDecisionAPIError()
+    {
+        $response = new Response(400, []);
+        $responses = [
+            $response,
+        ];
+
+        $client = $this->getClient($responses);
+
+        // Get Capture Manager
+        $manager = $client->getCaptureManager();
+        try {
+            $manager->add($this->captures);
+        } catch (RequestException $e) {
+            $this->assertEquals($e->getResponse()->getStatusCode(), 400);
+        }
+    }
+}
```

Comment:
This is a PHPUnit test I think? A vanity way to assert exception would be using [@expectedException and its friends](https://phpunit.de/manual/current/en/appendixes.annotations.html#appendixes.annotations.expectedException). It is no biggie either way though.